src/webargs/multidictproxy.py

```diff
@@ -0,0 +1,67 @@
+from webargs.compat import MARSHMALLOW_VERSION_INFO, Mapping
+from webargs.core import missing, is_multiple
+
+
+class MultiDictProxy(Mapping):
+    """
+    A proxy object which wraps multidict types along with a matching schema
+    Whenever a value is looked up, it is checked against the schema to see if
+    there is a matching field where `is_multiple` is True. If there is, then
+    the data should be loaded as a list or tuple.
+
+    In all other cases, __getitem__ proxies directly to the input multidict.
+    """
+
+    def __init__(self, multidict, schema):
+        self.data = multidict
+        self.multiple_keys = self._list_multiple_keys(schema)
+
+    def _list_multiple_keys(self, schema):
+        result = []
+        for name, field in schema.fields.items():
+            if not is_multiple(field):
+                continue
+            if MARSHMALLOW_VERSION_INFO[0] < 3:
+                result.append(field.load_from or name)
```

Comment:
```suggestion
                result.append(field.load_from if field.load_from is not None else name)
```

Adding a self-suggestion so that I can batch this with the GitHub suggestion commit for the change below.